src/plugins/enum/README.md

```diff
@@ -0,0 +1,19 @@
+- infos = Information about the enum plugin is in keys below
+- infos/author = Thomas Waser <>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides = check
+- infos/placements = presetstorage
+- infos/description =
+
+## Introduction ##
+
+The Enum plugin checks string values of Keys by comparing it against a list of valid values.
+
+## Usage ##
+
+The plugin checks every Key in the Keyset for the Metakey `check/enum` containing a list with the syntax `['string1', 'string2', 'string3', ..., 'stringN']` and compares each value with the string value of the Key. If no match is found an error is returned.
```

Comment:
> The [] isn't needed,i was trying to adapt the syntax of the specification
> file.

The specification could be in any syntax as long as it is parsable by one of 
Elektras plugin capable of storing meta data.

We are now defining the syntax within the value, not the syntax that will be 
stored on the disc. You could easily write a plugin that wraps every value 
with `[]`.

> Basically every alphanumeric string divided by an non-alphanumeric
> character is considered as a valid element right now,

So currently only alphanumeric strings can be used within the enum?
Its quite restricting, but basically it would suffice for a first version.

> but i can change it to just use a comma as a valid seperator

Yes, its better to be strict in the beginning. We can relax it later on if its 
somehow needed without breaking anyones configuration.

> values in <> to keys

I am not sure what you mean here.

> and not quoted values to plugin configured values ?

This is would be an optional feature. If you do not have worries that it might 
be slow in the 700x700 situation, we do not need it for a first version.